File: src/fb2site/inspector.py
from pathlib import Path

from .models import Inspection

EXPECTED_DIRS = [
    "your_facebook_activity",
    "personal_information",
]


def inspect_export(path: Path) -> Inspection:
    result = Inspection()

    if not path.exists():
        result.errors.append(f"{path} does not exist")
        return result

    if not path.is_dir():
        result.errors.append(f"{path} is not a directory")
        return result

    result.directory_exists = True

    missing = [
        directory
        for directory in EXPECTED_DIRS
        if not (path / directory).exists()
    ]

    if missing:
        result.errors.extend(
            f"Missing directory: {directory}"
            for directory in missing
        )
        return result

    result.is_facebook_export = True
    result.export_type = "facebook"

    return result

File: app/src/main/kotlin/in/sartaj/fb2site/inspect/Inspector.kt
package `in`.sartaj.fb2site.inspect

import `in`.sartaj.fb2site.model.Inspection
import java.nio.file.Files
import java.nio.file.Path

private val EXPECTED_DIRS = listOf(
    "your_facebook_activity",
    "personal_information"
)

fun inspectExport(path: Path): Inspection {
    val result = Inspection()

    if (!Files.exists(path)) {
        result.errors.add("$path does not exist")
        return result
    }

    if (!Files.isDirectory(path)) {
        result.errors.add("$path is not a directory")
        return result
    }

    result.inputExists = true

    val missing = EXPECTED_DIRS.filter {
        !Files.exists(path.resolve(it))
    }

    if (missing.isNotEmpty()) {
        result.errors.addAll(
            missing.map { "Missing directory: $it" }
        )
        return result
    }

    result.isFacebookExport = true
    result.exportType = "facebook"

    return result
}
